source/game/system/KPadController.cc

```diff
@@ -110,76 +111,75 @@ void RaceInputState::reset() {
     trickRaw = 0;
 }
 
+bool RaceInputState::accelerate() const {
+    return !!(buttons & 0x1);
+}
+
+bool RaceInputState::trickUp() const {
+    return !!(trick & 0x10);
+}
+
 KPadGhostButtonsStream::KPadGhostButtonsStream()
-    : buffer(nullptr), sequenceCount(-1), size(RKG_INPUT_DATA_SIZE), state(2) {}
+    : currentSequence(std::numeric_limits<u32>::max()), state(2) {}
 
 KPadGhostButtonsStream::~KPadGhostButtonsStream() = default;
 
-bool KPadGhostButtonsStream::readIsNewSequence(const u16 *sequence) const {
-    return readSequenceFrames >= (*sequence >> 0x8);
+bool KPadGhostButtonsStream::readIsNewSequence() const {
+    return readSequenceFrames >= (currentSequence >> 0x8);
 }
 
-u16 KPadGhostButtonsStream::readVal(const u16 *sequence) const {
-    return *sequence;
+u8 KPadGhostButtonsStream::readVal() const {
+    return currentSequence & 0xFF;
 }
 
-u16 KPadGhostButtonsStream::readFrame() {
-    constexpr u32 SEQUENCE_SIZE = 0x2;
-
+u8 KPadGhostButtonsStream::readFrame() {
     if (state != 1) {
         return 0;
     }
 
-    bool bVar1 = false;
-    const u16 *sequence = nullptr;
-
-    if (sequenceCount >= 0) {
-        sequence = buffer + sequenceCount;
-
-        if (readIsNewSequence(sequence)) {
-            bVar1 = true;
-            if (size <= ++sequenceCount * SEQUENCE_SIZE) {
+    if (currentSequence == std::numeric_limits<u32>::max()) {
+        readSequenceFrames = 0;
+        currentSequence = parse<u16>(buffer.read_u16());
+    } else {
+        if (readIsNewSequence()) {
+            readSequenceFrames = 0;
+            if (buffer.eof()) {
                 state = 2;
+            } else {
+                currentSequence = parse<u16>(buffer.read_u16());
             }
         }
-    } else {
-        sequenceCount = 0;
-        bVar1 = true;
     }
 
     if (state != 1) {
         return 0;
     }
 
-    sequence = buffer + sequenceCount;
-
-    if (bVar1) {
-        readSequenceFrames = 0;
-    }
-
     ++readSequenceFrames;
 
-    return readVal(sequence);
+    return readVal();
 }
 
-KPadGhostFaceButtonsStream::KPadGhostFaceButtonsStream() = default;
+KPadGhostFaceButtonsStream::KPadGhostFaceButtonsStream() : KPadGhostButtonsStream() {}
 
 KPadGhostFaceButtonsStream::~KPadGhostFaceButtonsStream() = default;
 
-KPadGhostDirectionButtonsStream::KPadGhostDirectionButtonsStream() = default;
+KPadGhostDirectionButtonsStream::KPadGhostDirectionButtonsStream() : KPadGhostButtonsStream() {}
 
 KPadGhostDirectionButtonsStream::~KPadGhostDirectionButtonsStream() = default;
 
-KPadGhostTrickButtonsStream::KPadGhostTrickButtonsStream() = default;
+KPadGhostTrickButtonsStream::KPadGhostTrickButtonsStream() : KPadGhostButtonsStream() {}
```

Comment:
Oops, messed this up during merge conflict resolution. This should go back to using `= default`